library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.util.DisplayMetrics
+import androidx.annotation.MainThread
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.internal.AsyncImageProcessingCallback
+import com.datadog.android.sessionreplay.internal.utils.Base64Utils
+import com.datadog.android.sessionreplay.internal.utils.DrawableUtils
+import com.datadog.android.sessionreplay.model.MobileSegment
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+@Suppress("UndocumentedPublicClass")
+class Base64Serializer private constructor(
+    private val threadPoolExecutor: ExecutorService,
+    private val drawableUtils: DrawableUtils = DrawableUtils(),
+    private val base64Utils: Base64Utils = Base64Utils,
+    private val webPImageCompression: ImageCompression = WebPImageCompression()
+) {
+
+    private var asyncImageProcessingCallback: AsyncImageProcessingCallback? = null
+
+    // region internal
+
+    @MainThread
+    internal fun handleBitmap(
+        displayMetrics: DisplayMetrics,
+        drawable: Drawable,
+        imageWireframe: MobileSegment.Wireframe.ImageWireframe
+    ) {
+        asyncImageProcessingCallback?.startProcessingImage()
+
+        val bitmap = drawableUtils.createBitmapFromDrawable(drawable, displayMetrics)
+
+        if (bitmap == null) {
+            asyncImageProcessingCallback?.finishProcessingImage()
```

Comment:
why do you call `finishProcessingImage` here and also in the `finalizeRecordedDataItem` ?

Reply:
In order that if we failed to get the bitmap we won't leave the item stuck in the queue. In `finalizeRecordedDataItem ` we follow the path where we successfully got the bitmap

Reply:
oh my bad nevermind